src/fallout.coffee

```diff
@@ -63,15 +60,12 @@
     onFalloutStoreChange: ->
       for name, selector of @_falloutWatchSelectors()
         storeValue = Fallout.store.data.getIn selector
-        unless @state["_#{name}"] is storeValue
+        unless Immutable.is @state["_#{name}"], storeValue
```

Comment:
Ah yeah, you might not need Immutable here in that case. Depends on whether or not more comparison checks pay off with fewer setState() calls

On the bright side, looks like Immutable.is first does an === comparison and short-circuits if true before proceeding to its deep 'valueOf' compare.
https://github.com/facebook/immutable-js/blob/master/src/is.js#L65